### DISCUSSION TITLE
Phalcon 5 Migration: Service 'router' was not found in the dependency injection container

### DISCUSSION THREAD

**Discussion by casparjones** [phalcon/phalcon#259] [Q&A]
Guten Tag,
I am currently trying to upgrade a Phalcon 4 app to Phalcon5 RC4.
Unfortunately I get in my app with the call:
`echo $application->handle($_SERVER["REQUEST_URI"])->getContent();`
the error:
`Service 'router' was not found in the dependency injection container`

Do I need to register the router as a service?

Best regards Frank

**Comment by niden** [comment]
If you are using the `FactoryDefault` then you don't need to set it up. If not you will need to. 

Also, this repository is for v6 our PHP based version. A lot of objects from the v5 are not implemented and as such if you are using this repository, the router has not been implemented yet.
